subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/results/DefaultVisitedGraphResults.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.results;
+
+import org.gradle.api.Describable;
+import org.gradle.api.artifacts.ResolveException;
+import org.gradle.api.artifacts.UnresolvedDependency;
+import org.gradle.api.internal.artifacts.result.MinimalResolutionResult;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Consumer;
+
+/**
+ * Default implementation of {@link VisitedGraphResults}.
+ */
+public class DefaultVisitedGraphResults implements VisitedGraphResults {
+
+    private final Describable resolveContextDisplayName;
+    private final MinimalResolutionResult resolutionResult;
+    private final Set<UnresolvedDependency> unresolvedDependencies;
+    private final ResolveException additionalFailure;
```

Comment:
❓  I still feel uncomfortable with the vagueness of how we refer to these failures.  What causes an "additional" failure?  A dependency that could successfully resolve, but we have a rule that prevents us from selecting a resolvable version?  Just taking a quick look, the only place I could find this being populated is for conflict resolution errors.  Are there other places where we set this?  If not, should we just call this `conflictResolutionFailures` or `dependencyConflictFailures` or something like that?  If there are other things we use this for, maybe we could call these more generally `dependencySelectionFailures` or something like that?

Reply:
Fatal failures caught by the `ErrorHandlingConfigurationResolver` also appear here. Further, I would eventually like to move dependency-locking failures here. They are currently modeled as unresolved dependency edges, but they were never part of the graph to start with and IMO shouldn't appear like that

In general, this is a bucket containing _any_ failure that is not visible from a resolution result.

Reply:
We met and decided renaming is the way to go here